src/clj/harja/kyselyt/yksikkohintaiset_tyot.sql

```diff
@@ -218,7 +218,7 @@ FROM toteuma tot
 WHERE tot.urakka IN (SELECT id
                      FROM urakka
                      WHERE hallintayksikko = :hallintayksikko
-                           AND (:urakkatyyppi::urakkatyyppi IS NULL OR tyyppi = :urakkatyyppi::urakkatyyppi)
+                           AND (TRUE IN (SELECT UNNEST(ARRAY[:urakkatyyppi]::urakkatyyppi[]) IS NULL) OR tyyppi = ANY(ARRAY[:urakkatyyppi]::urakkatyyppi[]))
```

Comment:
Eikös tuo entinen `:urakkatyyppi::urakkatyyppi IS NULL` ole ihan toimiva ja selkeämpi kuin `(TRUE IN (SELECT UNNEST(ARRAY[:urakkatyyppi]::urakkatyyppi[]) IS NULL)`, jos tarkoituksena on kattoa, että onko toi `:urakkatyyppi` `NULL`?

Reply:
Itsekin ajattelin just noin viime iltana. Kokeilin sitten tänään fiksata tuon entiselleen, mutta ei enää toiminut. 
Mietin, että sehän johtuu juurikin tästä:
```
urakkatyyppi (when urakkatyyppi
                        (case urakkatyyppi
                          :hoito ["hoito" "teiden-hoito"]
                          [(name urakkatyyppi)]))
```

* jos urakkatyyppi annetaan -> muokataan se sopivaksi ja käytetään queryssä
* Jos ei, se on NIL

Jos käytetään vanhaa vertailutyyppiä tuon `TRUE IN` ehdon sijaan, saadaan aikaiseksi query:

```sql
WHERE hallintayksikko = :hallintayksikko
AND (:urakkatyyppi::urakkatyyppi IS NULL OR tyyppi = ANY(ARRAY[:urakkatyyppi]::urakkatyyppi[]))
```

Tuossahan `:urakkatyyppi` parametria ei käsitellä samassa tyypissä. Toinen on määritelty arrayksi ja toinen taas ei.

Jos tuohon passataan `nil`, query preparoidaan ja ajetaan:
```sql
...AND (null::urakkatyyppi IS NULL OR tyyppi = ANY(ARRAY[null]::urakkatyyppi[]))...
```
Tämä toimii, mutta jos passataan vektori ['hoito' 'teiden-hoito']: 
```sql
... AND (['hoito','teiden-hoito']::urakkatyyppi IS NULL OR tyyppi = ANY(ARRAY['hoito','teiden-hoito']::urakkatyyppi[]))...
````
Tämä sitten aiheuttaa failin.

Mergesin tämän muutoksen as-is, kun se toimii kaikissa em. tilanteissa. Tuota voi toki jälkikäteenkin fiksata, jos näyttää bloatilta.